tests/testthat/test-infrastructure.r

```diff
@@ -59,3 +59,68 @@ test_that("use_data", {
     expect_equal(local_test_data_item_to_save, 42L)
   })
 })
+
+test_that("use_package", {
```

Comment:
With the exception of a change to [L75](https://github.com/hadley/devtools/pull/1054/files#diff-67b82d9b2f120c72b1173ce45fcdb8e7R75), I don't really see how this is "overkill" so much as it's just thorough. Every potential use of the function is tested

1. for proper functionality
2. for idempotency 

and finally a check to make sure the docs aren't lying and an error is thrown on an invalid version number is performed.

If this is, indeed, overkill, which tests would you keep?